domains/transfer-agreements/TransferAgreements.API/API/Filters/AuditDotNetFilter.cs

```diff
@@ -0,0 +1,30 @@
+using System.Security.Claims;
+using System.Threading.Tasks;
+using API.Extensions;
+using Audit.Core;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace API.Filters;
+
+public class AuditDotNetFilter : IAsyncActionFilter
+{
+    public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+    {
+        var httpContext = context.HttpContext;
+        var claimsPrincipal = httpContext.User;
+
+        var actorId = claimsPrincipal.FindFirstValue("actor") ?? claimsPrincipal.FindFirstValue("atr");
```

Comment:
Why is a fallback value needed?

Reply:
The jwt token, in the demo environment, has 2 different claims, named "actor" and "atr". both contain the actor. So in the case that one of them were removed (most like actor, since the other claims are 3 letters in name), i created this fallback.

Though, this is no longer relevant, since it has been changed to make use of the HttpUserExtension class methods now.